README.md

```diff
@@ -1,108 +1,298 @@
+Kubernetes controller before).
+
+[^2]: In order to cover providers that use Gardnener, a generic Crossplane
+"Composite Resource Definition" like in [^1] may be created. Alternatively,
+Gardnener CRD's could maybe just be mirrored in their Central API instance,
+still creating an interoperability benefit through "semantic" compatibility.
+
+[^3]: Which is also not to say that it will be suggested to providers to disable
+their public OpenStack/Keycloak/... API's, preventing use of native
+OpenStack/Keycloak/... tooling and breaking existing solutions.
+Extensively using these API's together with the central API may compromise
+the benefits of its uniform AuthZ, though.
+
+### Kubernetes API
+
+Instead of creating SCS-specific API idioms and building the implementation
+from scratch, the Kubernetes API will be "reused". Essentially, the Kubernetes
+API is just an opinionated REST API which has opinions on how a resource
+is defined, how it looks like, how it is reconciled/handled, how AuthN/AuthZ
+can be implemented. The Kubernetes ecosystem provides much tooling for working
+with such (custom) resource definitions: For creating the definitions
+themselves, building controllers, making them discoverable and deployable.
+
+As such, Kubernetes is a great choice for building any sort of resource
+management API - with some caveats regarding its deployment and the legacy
+of starting off as container orchestration tooling.
+
+### Crossplane tooling
+
+Crossplane even extends the Kubernetes API with
+"[Compositions](https://docs.crossplane.io/v1.14/concepts/compositions/)" and
+"[Composite Resource Definitions](https://docs.crossplane.io/v1.14/concepts/composite-resource-definitions/)"
+(XRD) to make Kubernetes the base for platform engineering within organizations.
+
+Secondly, it provides an API machinery to bring any cloud resource into Kubernetes
+using backend-specific "providers" (roughly comparable with Terraform providers).
+As such, Crossplane with its provider ecosystem actually already did most of
+the heavy lifting for providing e.g. OpenStack or Keycloak resources inside of Kubernetes.
+
+On top, the platform engineering concepts in Crossplane make building multi-tenancy
+systems pretty straight-forward, even for
+[single clusters](https://docs.crossplane.io/knowledge-base/guides/multi-tenant/#single-cluster-multi-tenancy).
+
+Alright. Crossplane takes care of exposing OpenStack resources and does some
+fancy stuff regarding multi-tenancy. What about providing actual Kubernetes
+**workload** clusters?
+
+### Cluster stacks / Cluster API
+
+[Cluster stacks](https://github.com/SovereignCloudStack/cluster-stacks) do
+[not replace the use of Cluster API](https://github.com/SovereignCloudStack/cluster-stack-operator/blob/adb648ceaebddca04a015fbea0319110ca99a5cc/docs/architecture/user-flow.md#recap---how-do-cluster-api-and-cluster-stacks-work-together).
+Instead, they are complementing Cluster API by providing `ClusterClasses`.
```

Comment:
... as well as node images (if required) and cluster addons in workload clusters!

Reply:
For the purposes of this document, I saw node images and cluster addons more like something that a `ClusterClass` provides/embeds.

But, sure, mentioning it does not hurt. Created 414b773.

Reply:
no clusterclass is definitely not related to the cluster addons. ok!